lib/active_model_serializers/lookup_chain.rb

```diff
@@ -0,0 +1,43 @@
+module ActiveModelSerializers
+  module LookupChain
+    BY_RESOURCE = lambda do |_resource_class, resource_class_name, _serializer_class|
+      "#{resource_class_name}Serializer"
+    end
+
+    BY_NAMESPACE = lambda  do |resource_class, resource_class_name, _serializer_class|
+      resource_namespace = resource_class.name.deconstantize
+      serializer_name = "#{resource_class_name}Serializer"
```

Comment:
serializer_name is duplicated, and so is resource_namespace.

This is why I'm more leaning towards this NOT being an array of procs, so that we can keep the code duplication down.